### PR TITLE
Copy Locale management UI from Wagtail core

### PR DESCRIPTION
**Pull request by kaedroho**
We need to copy it in order to implement Locale components. Hopefully these will make it into Wagtail 2.12, but for now we can support them with a custom locale management UI.

### CODE REVIEW

File: wagtail_localize/locales/wagtail_hooks.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin.menu import MenuItem
+from wagtail.core import hooks
+from wagtail.core.permissions import site_permission_policy
+
+from .views import LocaleViewSet
+
+
+@hooks.register('register_admin_viewset')
```
Comment (KalobTaulien) [inline wagtail_localize/locales/wagtail_hooks.py:12]: If this hook is in core, is there any way I could convince you to add this to the Wagtail docs?

**Review by KalobTaulien — APPROVED** [review]
Only one TODO that should probably be addressed, but it's non-blocking imo so this gets a 👍

File: wagtail_localize/locales/locale/es/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# SOME DESCRIPTIVE TITLE.
+# Copyright (C) YEAR THE PACKAGE'S COPYRIGHT HOLDER
+# This file is distributed under the same license as the PACKAGE package.
+# FIRST AUTHOR <EMAIL@ADDRESS>, YEAR.
+#
+# Translators:
+# Maylon Pedroso <maylonpedroso@gmail.com>, 2020
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_localize/locales/locale/es/LC_MESSAGES/django.po:7]: This is exactly how it is in Wagtail

File: wagtail_localize/locales/templates/wagtaillocales/index.html [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                                <div class="title-wrapper">
+                                    <a href="{% url 'wagtaillocales:edit' locale.id %}">{{ locale }}</a>
+                                    {% if not locale.language_code_is_valid %}
+                                        {% trans "This locale's language code is not supported" as error %}
+                                        {% icon name="warning" class_name="locale-error" title=error %}
+                                    {% endif %}
+                                </div>
+                            </td>
+                            <td>
+                                {# TODO Make this translatable #}
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail_localize/locales/templates/wagtaillocales/index.html:37]: Probably should fix this in Wagtail core, this copy is only temporary

File: wagtail_localize/locales/wagtail_hooks.py (hunk 1 above)
Comment (kaedroho) [reply wagtail_localize/locales/wagtail_hooks.py:12]: I think the plan was to develop it a bit more into an official API first. That was a few years ago!

Hopefully we will have a chance to revisit this in the platform sub-team

**Comment by kaedroho** [comment]
Thanks!

**Pull request merged**